src/users/users.controller.ts

```diff
@@ -376,4 +379,32 @@ export class UsersController {
       telegram: dto.telegram,
     });
   }
+
+  @ApiExcludeEndpoint()
+  @Post(':id/token')
+  @UseGuards(ApiKeyGuard)
+  async createAuthToken(
+    @Param('id', new IntIsSafeForPrismaPipe())
+    id: number,
+  ): Promise<SerializedUserLoginUrl> {
+    const user = await this.usersService.findOrThrow(id);
+
+    const secret: Secret = this.config.get<string>('JWT_TOKEN_SECRET');
+    const options: SignOptions = {
+      algorithm: 'HS256',
+      expiresIn: '1 days',
+    };
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call
+    const token: string = sign(
+      { sub: user.email, iat: Date.now() },
+      secret,
+      options,
+    );
+
+    return {
+      url: `https://api.ironfish.network/login/${token}`,
+      email: user.email,
```

Comment:
Do we need to return the email?

Reply:
Its for support. When they get the login url, they can also use the email address to send email.
They could get the email from retool by user id. I am fine not returning the email.

Reply:
Yeah sure, that seems fine